src/storage/compactor/src/server.rs

```diff
@@ -81,7 +82,7 @@ pub async fn prepare_start_parameters(
         &system_params_reader,
         &storage_memory_config,
     )));
-    let non_reserved_memory_bytes = (system_memory_available_bytes() as f64
+    let non_reserved_memory_bytes = (compactor_opts.compactor_total_memory_bytes as f64
```

Comment:
Compactor opts provide an instance of `compactor_total_memory_bytes`, but inside `RwConfig`, there's a separate value:
```
    pub compactor_memory_limit_mb: Option<usize>,
```

What's the difference between these configurations? Will the one in RwConfig always be ignored, if so can we remove it?

Reply:
Good question:

I found that compactor_memory_limit_mb is only used in embedding compactor mode and occupies part of the CN's memory (you can refer to the storage_memory_config function).

I think this is different from standalone deployment mode, I didn't change it in this PR, maybe you have a better idea?

Reply:
> Good question:
> I found that compactor_memory_limit_mb is only used in embedding compactor mode and occupies part of the CN's memory (you can refer to the storage_memory_config function).
> I think this is different from standalone deployment mode, I didn't change it in this PR, maybe you have a better idea?

Got your point. It's out of scope for this PR then. We can just rename it.
IIRC embedded compactor is not used for standalone deployment.